### PR TITLE
Examples and test cleanup

### LINKED ISSUE #1436: De-duplicate and generally improve examples
Prior to HIL testing, our examples were serving a dual purpose, both demonstrating use of the HAL and testing/verifying the functionality. With HIL testing now available, we should focus our examples on using the hardware, not proving out the drivers.

There are also a number of examples specific to third-party trait packages, e.g. all the `eh1_` prefixed examples. We should rework our examples to demonstrate only the applicable driver(s), the trait packages should be irrelevant here.

Additionally, we should try to better comment and just generally improve examples as able.

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

### Pull Request Details 📖

Closes #1436

Opening a bit before finishing (I plan on polishing some of the remaining examples) but I wanted to get opinions on the examples I nuked, and the ones I've kept so far.

### CODE REVIEW

File: hil-test/tests/interrupt.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            csrr {x}, 0x7e2
+            ",
+            options(nostack),
+            x = inout(reg) perf_counter,
+        )
+    };
+    info!("Performance counter:{}", perf_counter);
+    cfg_if::cfg_if! {
+        if #[cfg(any(feature = "esp32c3", feature = "esp32c2"))] {
+            assert!(perf_counter < 1000);
```
Comment (MabezDev) [inline hil-test/tests/interrupt.rs:82]: Interesting difference here, but I guess they are different interrupt controllers. We will also have a difference when we add a test for the Xtensa chips.

File: hil-test/tests/interrupt.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
+//! Delay Test
```
Comment (SergioGasquez) [inline hil-test/tests/interrupt.rs:1]:
```suggestion
//! Interrupt Test
```

**Comment by Dominaezzz** [comment]
Are the deleted examples going to become tests? Otherwise I don't see benefit in losing them.

The eh1 SPI examples, aren't they useful to demonstrate how to get from a HAL type to an eh1 trait object? (I'm a bit biased since I'm trying to avoid merge conflicts :sweat_smile:)
Or perhaps this is something that should be a snippet in the doc.

**Comment by MabezDev** [comment]
> Are the deleted examples going to become tests? Otherwise I don't see benefit in losing them.

Most already are, at least the crypto stuff is tested, and I converted the direct vectoring example to an interrupt latency test. The other examples IMO do not add that much value, I don't personally see the value in an example that does the same as another but imports eh to do so.

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

**Comment by jessebraham** [comment]
This test seems pretty flaky, maybe we should skip it for the time being, just for the sake of getting the other changes merged?

**Comment by MabezDev** [comment]
The thing is I'm quite confused as to why there is more latency than I see running it locally :thinking:. I thought about doing a few runs, but actually, the first-time latency is an important metric in its own right.

I think I'll just set a high enough number for now so that it passes; it will still catch interrupts not working at all for example as the test will time out.

**Pull request merged**